torch/lib/THCS/generic/THCSTensorMath.cu

```diff
@@ -463,6 +463,38 @@ void THCSTensor_(cmul)(THCState *state, THCSTensor *r_, THCSTensor *t, THCSTenso
   THCTensor_(free)(state, s_values_);
 }
 
+#if defined(THCS_REAL_IS_FLOAT) || defined(THCS_REAL_IS_DOUBLE)
+void THCSTensor_(pow)(THCState *state, THCSTensor *r_, THCSTensor *t, real value) {
```

Comment:
It seems that all pointwise functions will have nearly the same code. Maybe it's worth extracting to some helpers or macros?